crates/ruff_python_ast/src/visitor/source_order.rs

```diff
@@ -6,8 +6,10 @@ use crate::{
 };
 use crate::{AnyNodeRef, AstNode};
 
-/// Visitor that traverses all nodes recursively in pre-order.
-pub trait PreorderVisitor<'a> {
+/// Visitor that traverses all nodes recursively in the order they appear in the source.
+///
+/// If you need a visitor that visits the nodes in their semantic order, use [`Visitor`](super::Visitor) instead.
```

Comment:
nit: I'd find this slightly clearer

```suggestion
/// If you need a visitor that visits the nodes in the order they're evaluated at runtime,
/// use [`Visitor`](super::Visitor) instead.
```

Would it also be worth renaming the other one to `EvaluationOrderVisitor`?

Reply:
I don't think renaming the other one is really necessary; evaluation order is what I would expect by default (that's what CPython's AST visitor does, for example.)

Reply:
Right, I wondered if it would make distinguishing the two slightly clearer, but I agree it's what I'd expect by default